### PR TITLE
Orion Rocky 9 updates

### LINKED ISSUE #1159: Update Orion modulefiles to Rocky 9
Received the following from RDHPCS Management

> Orion’s Operating System (OS) and software stack is scheduled to be upgraded during a two day downtime, starting on Wednesday, June 12th and going through Thursday, June 13th. The OS on Orion will be upgraded from CentOS 7 to Rocky 9, another derivative of Red Hat Linux.

This issue is opened to document the updating of `modulefiles/EVA/orion.lua` and `modulefiles/GDAS/orion.intel.lua`   to Rocky 9

### PR DESCRIPTION
**Pull request by RussTreadon-NOAA**
Orion now runs Rocky 9.  This upgrade requires changes to how GDASApp builds and runs on Orion.  This PR adds Orion Rocky 9 updates to GDASApp

Fixes #1159

### CODE REVIEW

File: test/soca/gw/CMakeLists.txt [hunk 1]
```diff
@@ -10,25 +10,14 @@ add_test(NAME test_gdasapp_soca_prep
...
-    set(MACHINE "hercules")
-    set(PARTITION "hercules")
-  ELSE()
-    set(MACHINE "orion")
-    set(PARTITION "orion")
-  ENDIF()
-ENDIF()
-IF (IS_DIRECTORY /scratch2/NCEPDEV/)
-  set(MACHINE "hera")
+if (MACHINE STREQUAL "hercules")
```
Comment (guillaumevernieres) [inline test/soca/gw/CMakeLists.txt:13]: Ha! I was going to work on this today ... Thanks for the fix @RussTreadon-NOAA !

**Comment by RussTreadon-NOAA** [comment]
The changes in this PR were manually added to g-w PR [#2700](https://github.com/NOAA-EMC/global-workflow/pull/2700).  GDASApp ctests with g-w were [run](https://github.com/NOAA-EMC/global-workflow/pull/2700#issuecomment-2180691132).  48 out of 48 tests pass.

**Comment by RussTreadon-NOAA** [comment]
This PR is ready for review given the successful [testing](https://github.com/NOAA-EMC/global-workflow/pull/2700#issuecomment-2180691132) performed in g-w PR [#2700](https://github.com/NOAA-EMC/global-workflow/pull/2700)

File: test/soca/gw/CMakeLists.txt (hunk 1 above)
Comment (RussTreadon-NOAA) [reply test/soca/gw/CMakeLists.txt:13]: @guillaumevernieres , if you are OK with the changes in this PR I can merge pending a review & approval

**Review by guillaumevernieres — APPROVED** [review]

File: test/soca/gw/CMakeLists.txt (hunk 1 above)
Comment (guillaumevernieres) [reply test/soca/gw/CMakeLists.txt:13]: Yes, looks good @RussTreadon-NOAA . Thanks again.

**Pull request merged**